src/server/scripts/Spells/spell_druid.cpp

```diff
@@ -197,6 +204,72 @@ class spell_dru_enrage : public SpellScriptLoader
         }
 };
 
+// 37336 - Druid Forms Trinket
+class spell_dru_forms_trinket : public SpellScriptLoader
+{
+public:
+    spell_dru_forms_trinket() : SpellScriptLoader("spell_dru_forms_trinket") { }
+
+    class spell_dru_forms_trinket_AuraScript : public AuraScript
+    {
+        PrepareAuraScript(spell_dru_forms_trinket_AuraScript);
+
+        bool Validate(SpellInfo const* /*spellInfo*/) override
+        {
+            if (!sSpellMgr->GetSpellInfo(SPELL_DRUID_FORMS_TRINKET_BEAR)
+                || !sSpellMgr->GetSpellInfo(SPELL_DRUID_FORMS_TRINKET_CAT)
+                || !sSpellMgr->GetSpellInfo(SPELL_DRUID_FORMS_TRINKET_MOONKIN)
+                || !sSpellMgr->GetSpellInfo(SPELL_DRUID_FORMS_TRINKET_NONE)
+                || !sSpellMgr->GetSpellInfo(SPELL_DRUID_FORMS_TRINKET_TREE))
+                return false;
+            return true;
+        }
+
+        void HandleProc(AuraEffect const* aurEff, ProcEventInfo& eventInfo)
+        {
+            PreventDefaultAction();
+            if (Unit* target = eventInfo.GetActionTarget())
+            {
+                uint32 triggerspell = 0;
+
+                switch (target->GetShapeshiftForm())
+                {
+                case FORM_BEAR:
+                case FORM_DIREBEAR:
+                    triggerspell = SPELL_DRUID_FORMS_TRINKET_BEAR;
+                    break;
+                case FORM_CAT:
+                    triggerspell = SPELL_DRUID_FORMS_TRINKET_CAT;
+                    break;
+                case FORM_MOONKIN:
+                    triggerspell = SPELL_DRUID_FORMS_TRINKET_MOONKIN;
+                    break;
+                case FORM_NONE:
+                    triggerspell = SPELL_DRUID_FORMS_TRINKET_NONE;
+                    break;
+                case FORM_TREE:
+                    triggerspell = SPELL_DRUID_FORMS_TRINKET_TREE;
+                    break;
+                default:
+                    return;
```

Comment:
but you still have a list of valid shapeshifts (remember that there are forced shapeshift spells apart from player known ones)